geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseIndexAdapter.scala

```diff
@@ -554,18 +580,19 @@ object HBaseIndexAdapter extends LazyLogging {
   }
 
   /**
-    * Writer for hbase
-    *
-    * @param ds datastore
-    * @param indices indices to write to
-    * @param partition partition to write to
-    */
+   * Writer for hbase
+   *
+   * @param ds        datastore
+   * @param indices   indices to write to
+   * @param partition partition to write to
+   */
   class HBaseIndexWriter(
-      ds: HBaseDataStore,
-      indices: Seq[GeoMesaFeatureIndex[_, _]],
-      wrapper: FeatureWrapper[WritableFeature],
-      partition: Option[String]
-    ) extends BaseIndexWriter(indices, wrapper) {
+                          ds: HBaseDataStore,
+                          indices: Seq[GeoMesaFeatureIndex[_, _]],
+                          wrapper: FeatureWrapper[WritableFeature],
+                          partition: Option[String],
+                          sft: Some[SimpleFeatureType]
```

Comment:
@jnh5y Points out we're changing the API for the HBaseIndexWriter. Is this find as is, should we wrap this in a compatibility layer or is there a better way to address this?

Reply:
i think this class is 'internal' so we can change it. i don't expect anyone would be instantiating it directly...
however, the feature indices have a reference to the sft, so you don't need to pass it in explicitly. you could instead do something like: `indices.headOption.flatMap(_.sft.getDtgIndex).getOrElse(-1)`

Reply:
Will the reference to sft always be there? I need to access `sft.getUserData`, but since that just returns data and not an Option, flatMap isn't playing nicely with it. It looks like I _could_ just use `indices.headOption.get.sft.getUserData`, but that feels like it'd break on some edge case that I can't think of at the moment.

Reply:
i think this may be moot if you use the helper function i mentioned below - but since user data is a java map, you'd have to wrap the result in an option to use flat map. So something like:
`indices.headOption.flatMap(i => Option(i.sft.getUserData.get("foo")))`

Reply:
Yep, once I got the other function working it's not an issue. In fact, as far as I can tell even dtgIndex becomes irrelevant because the schema definition specifies the attribute that's being used for TTL, so that simplifies things.